polygon/p2p/service.go

```diff
@@ -81,3 +87,34 @@ func (s *service) Penalize(ctx context.Context, peerId PeerId) error {
 func (s *service) ListPeersMayHaveBlockNum(blockNum uint64) []PeerId {
 	return s.peerTracker.ListPeersMayHaveBlockNum(blockNum)
 }
+
+// TODO(taratorio): move inside MessageListener
+func (s *service) OnNewBlock(ctx context.Context, callback func(*types.Block, PeerId)) {
```

Comment:
I try to add a test for every public function of the p2p.Service - please can you add a test in service_test.go?